### PR TITLE
Adding AST support to python clients

### PR DESCRIPTION
**Pull request by sarane22**
Sending AST params via custom configuations.
Params:
1. source_language
2. target_language
3. task

### CODE REVIEW

File: riva/client/argparse_utils.py [hunk 1]
```diff
@@ -85,6 +85,24 @@ def add_asr_config_argparse_parameters(
...
+        default="transcribe",
+        type=str,
+        help="Task for the model (transcribe/translate)",
+    )
+    parser.add_argument(
+        "--target-language",
+        default="",
+        type=str,
+        help="Target language for translation",
+    )
```
Comment (rmittal-github) [inline riva/client/argparse_utils.py:105]: we should support these command line params through the generic --metadata field we have in the client. Adding dedicated translation specific params (source/target language, task) can be confusing for users in the ASR client.

Comment (sarane22) [reply riva/client/argparse_utils.py:93]: Updated to use custom_configuration

File: riva/client/argparse_utils.py [hunk 2]
```diff
@@ -85,6 +85,12 @@ def add_asr_config_argparse_parameters(
         type=float,
         help="Threshold value for likelihood of blanks before detecting end of utterance",
     )
+    parser.add_argument(
+        "--custom-configuration",
+        default="",
+        type=str,
+        help="Add custom configurations to be sent to the custom backends. <key:value,key:value>",
```
Comment (rmittal-github) [inline riva/client/argparse_utils.py:92]: similar comment here for clarity, <key:value,key:value,...>

File: riva/client/asr.py [hunk 3]
```diff
@@ -152,6 +153,22 @@ def add_endpoint_parameters_to_config(
     inner_config.endpointing_config.CopyFrom(endpointing_config)
 
 
+def add_custom_configuration(
+    config: Union[rasr.RecognitionConfig, rasr.EndpointingConfig],
```
Comment (rmittal-github) [inline riva/client/asr.py:157]: shouldn't have `rasr.EndpointingConfig` here, right? probably a leftover typo

File: scripts/asr/transcribe_file_offline.py [hunk 4]
```diff
@@ -39,14 +39,18 @@ def main() -> None:
...
-        config,
-        args.start_history,
-        args.start_threshold,
-        args.stop_history,
-        args.stop_history_eou,
+        config, 
+        args.start_history, 
+        args.start_threshold, 
+        args.stop_history, 
+        args.stop_history_eou,
```
Comment (rmittal-github) [inline scripts/asr/transcribe_file_offline.py:46]: trailing whitespaces can be stripped

Comment (sarane22) [reply scripts/asr/transcribe_file_offline.py:46]: We should configure pre-commit for clients.

**Review by rmittal-github — DISMISSED** [review]

File: scripts/asr/transcribe_file_offline.py (hunk 4 above)
Comment (sarane22) [reply scripts/asr/transcribe_file_offline.py]: Updated

File: riva/client/asr.py (hunk 3 above)
Comment (sarane22) [reply riva/client/asr.py]: Updated

File: riva/client/argparse_utils.py (hunk 2 above)
Comment (sarane22) [reply riva/client/argparse_utils.py]: Updated

File: riva/client/__init__.py [hunk 5]
```diff
@@ -12,6 +12,7 @@
     print_streaming,
     sleep_audio_length,
     add_endpoint_parameters_to_config,
+    add_custom_configuration,
```
[the remark this answers is not in the document]
Comment (sarane22) [reply riva/client/__init__.py]: Updated

File: riva/client/argparse_utils.py [hunk 6]
```diff
@@ -85,6 +85,12 @@ def add_asr_config_argparse_parameters(
         type=float,
         help="Threshold value for likelihood of blanks before detecting end of utterance",
     )
+    parser.add_argument(
+        "--custom-configuration",
+        default="",
+        type=str,
+        help="Add custom configurations to be sent to the custom backends. <key:value,key:value,...>",
```
Comment (rmittal-github) [inline riva/client/argparse_utils.py:92]:
```suggestion
        help="Custom configurations to be sent to the server as key value pairs <key:value,key:value,...>",
```

**Review by rmittal-github — APPROVED** [review]

**Pull request merged**